corehq/apps/cleanup/management/commands/swap_duplicate_xforms.py

```diff
@@ -0,0 +1,158 @@
+
+                total_lines = 0
+                for _ in lines:
+                    total_lines += 1
+                lines.seek(0)
+
+                for i, (domain, bad_xform_id) in enumerate(l.split() for l in lines):
+                    self._print_progress(i, total_lines)
+                    duplicates = self.get_duplicates(domain, bad_xform_id)
+                    if len(duplicates) == 1:
+                        num_with_1_dup += 1
+                        self.swap_doc_types(log_file, bad_xform_id, list(duplicates)[0], dry_run)
+                    elif len(duplicates) > 1:
+                        num_with_multi_dup += 1
+                        self.log_too_many_dups(log_file, bad_xform_id, duplicates)
+                    else:
+                        num_with_no_dup += 1
+
+        print "Found {} forms with no duplicates".format(num_with_no_dup)
+        print "Found {} forms with one duplicate".format(num_with_1_dup)
+        print "Found {} forms with multiple duplicates".format(num_with_multi_dup)
+
+    def get_duplicates(self, domain, xform_id):
+
+        if domain not in self.dups_by_domain:
+            self.populate_dup_map(domain)
+        return self.dups_by_domain[domain].get(UUID(xform_id), {})
+
+    def populate_dup_map(self, domain):
+        self.dups_by_domain[domain] = defaultdict(set)
+        dups = XFormInstance.view(
+            'couchforms/all_submissions_by_domain',
+            startkey=[domain, 'XFormDuplicate', '2016'],
+            endkey=[domain, "XFormDuplicate", {}],
+            reduce=False,
+            include_docs=True
+        )
+        for dup in dups:
+            match = re.match('Form is a duplicate of another! \((.*)\)', dup.problem or "")
+            if match:
+                orig_id = match.groups()[0]
+                try:
+                    orig_id = UUID(orig_id)
+                except ValueError:
+                    continue
+                self.dups_by_domain[domain][orig_id].add(dup._id)
+
+    def swap_doc_types(self, log_file, bad_xform_id, duplicate_xform_id, dry_run):
+        bad_xform = XFormInstance.get(bad_xform_id)
+        duplicate_xform = XFormInstance.get(duplicate_xform_id)
+
+        # Convert the XFormInstance to an XFormDuplicate
+        bad_xform.doc_type = XFormDuplicate.__name__
+        bad_xform.problem = "Form was missing multimedia attachments. Replaced by {} on {}".format(
+            duplicate_xform_id, datetime.now().isoformat()
+        )
+        bad_xform = XFormDuplicate.wrap(bad_xform.to_json())
+
+        # Convert the XFormDuplicate to an XFormInstance
+        del duplicate_xform.problem
+        duplicate_xform.doc_type = XFormInstance.__name__
+        duplicate_xform = XFormInstance.wrap(duplicate_xform.to_json())
+
+        self.log_swap(log_file, bad_xform_id, duplicate_xform_id, dry_run)
+
+        if not dry_run:
+            duplicate_xform.save()
+            bad_xform.save()
```

Comment:
bulk saving would be a little more efficient, but the script doesn't take too long, and it would be annoying to replicate the pre-save behavior in `XFormInstance.save()`